lib/chkhash.c

```diff
@@ -0,0 +1,64 @@
+#include <regex.h>
+#include <string.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+
+/*
+ * match_regex - return true if mathch, false if not
+ */
+bool 
+match_regex(const char *pattern, const char *string) 
+{
+	regex_t regex;
+	if (regcomp(&regex, pattern, REG_EXTENDED) != 0)
+		return false;
+
+	int result = regexec(&regex, string, 0, NULL, 0);
+	regfree(&regex);
+	return result == 0;
+}
+
+/*
+ * is_valid_hash - check if the given string is a valid password hash
+ *
+ * Returns true if the string appears to be a valid hash, false otherwise.
+ */
+bool 
+is_valid_hash(const char *hash) 
+{
+	if (hash == NULL)
+		return false;
+
+	// Minimum length for DES
+	if (strlen(hash) <= 13)
+		return false;
+
+	// DES: exactly 13 characters from [A-Za-z0-9./]
+	if (strlen(hash) == 13 && match_regex("^[A-Za-z0-9./]{13}$", hash))
+		return true;
```

Comment:
Is the DES length at least 13 or exactly 13?

Reply:
Then I think the comment saying `Minimum length for DES" can be misleading.  Please remove it.